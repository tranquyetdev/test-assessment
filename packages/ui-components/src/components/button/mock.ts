import { ButtonProps } from './button';

export const mockButtonProps: Record<ButtonProps['type'], ButtonProps> = {
  button: {
    type: 'button',
    children: 'button',
    onClick: () => console.log('hello world'),
  },
  link: {
    type: 'link',
    children: 'link',
    href: 'https://github.com/',
    title: 'Github',
  },
};
