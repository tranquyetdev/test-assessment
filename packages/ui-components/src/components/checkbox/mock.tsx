import { CheckBoxProps, defaultBgcolor } from '.';

export const mockCheckBoxProps: CheckBoxProps = {
  label: 'Developer',
  bgColor: defaultBgcolor,
  id: 'checkbox',
  checked: false,
  disabled: false,
};
