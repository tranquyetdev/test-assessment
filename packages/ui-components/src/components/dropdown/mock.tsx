export const mockDropdownProps = {
  content: (
    <ul className="menu">
      <li className="menu-item">
        <button>Menu 1</button>
      </li>
      <li className="menu-item">
        <button>Menu 2</button>
      </li>
    </ul>
  ),
  button: <div className="border">Show content</div>,
};
