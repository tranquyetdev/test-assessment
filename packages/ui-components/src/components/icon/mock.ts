import { IconName, IconProps } from ".";

export const mockIcons: IconProps & { name?: IconName }[] = [
  { name: "arrow-left" },
  { name: "arrow-right" },
  { name: "check" },
  { name: "chevron-down" },
  { name: "remove" },
  { name: "edit" },
  { name: "notification" },
  { name: "plus" },
  { name: "publish" },
  { name: "qa" },
  { name: "refresh" },
  { name: "save" },
  { name: "search" },
  { name: "caution" },
  { name: "eye-hide" },
  { name: "eye-show" },
]