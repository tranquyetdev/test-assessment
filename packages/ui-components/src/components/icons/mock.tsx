import { IconName, IconProps } from ".";

export const mockIcons: IconProps & { name?: IconName }[] = [
  { name: "arrow-left" },
  { name: "arrow-right" },
  { name: "check" },
  { name: "chevron-down" },
  { name: "edit" },
  { name: "notification" },
  { name: "plus" },
  { name: "publish" },
  { name: "qa" },
  { name: "refresh" },
  { name: "remove" },
  { name: "save" },
  { name: "caution" },
  { name: "eye-hide" },
  { name: "eye-show" },
]