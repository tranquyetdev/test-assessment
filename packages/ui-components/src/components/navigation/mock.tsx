import { NavigationProps } from './types';

export const mockNavigationProps: NavigationProps = {
  menuItems: [
    {
      title: 'Test Management',
      url: '/test-management',
    },
    {
      title: 'Candidate',
      url: '/candidate',
    },
    {
      title: 'Candidate',
      url: '/candidate',
    },
  ],
  currentPathname: '/test-management',
};
