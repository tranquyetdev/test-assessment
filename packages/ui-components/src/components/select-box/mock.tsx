import { SelectBoxProps } from './types';

export const mockSelectBoxProps: SelectBoxProps = {
  options: [
    { name: 'option1', value: '1' },
    { name: 'option2', value: '2' },
    { name: 'option3', value: '3' },
    { name: 'option4', value: '4' },
    { name: 'option5', value: '5' },
  ],
  rightIcon: (
    <svg
      className="h-4 w-4"
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 20 20"
    >
      <path d="M9.293 12.95l.707.707L15.657 8l-1.414-1.414L10 10.828 5.757 6.586 4.343 8z" />
    </svg>
  ),
  placeholder: 'Select your option',
  label: 'Label',
  variant: 'verticle-label',
  size: 'small',
};
