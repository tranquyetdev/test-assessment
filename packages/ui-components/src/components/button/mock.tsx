import { RenderIcon } from '../icons';
import { ButtonProps } from './button';

export const mockButtonProps: Record<ButtonProps['type'], ButtonProps> = {
  button: {
    type: 'button',
    children: 'button',
    onClick: () => alert('hello world'),
    LeftIcon: <RenderIcon name="caution" className="text-error" />,
    RightIcon: (
      <RenderIcon name="caution" className="text-error" />
    ),
  },
  link: {
    type: 'link',
    children: 'link',
    href: 'https://github.com/',
    title: 'Github',
    LeftIcon: <RenderIcon name="caution" className="text-error" />,
    RightIcon: (
      <RenderIcon name="caution" className="text-error" />
    ),
  },
};
