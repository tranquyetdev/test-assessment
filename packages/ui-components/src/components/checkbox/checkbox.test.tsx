import React from 'react';
import { render } from '@testing-library/react';

import { CheckBox } from '.';
import { mockCheckBoxProps } from './mock';

describe('Checkbox', () => {
  it(`should render successfully`, () => {
    const checkBox = render(<CheckBox {...mockCheckBoxProps} />);
    expect(checkBox).toMatchSnapshot();
  });
});
