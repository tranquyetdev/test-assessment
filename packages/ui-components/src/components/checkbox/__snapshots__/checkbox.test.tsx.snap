// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render successfully 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex items-center"
      >
        <input
          class="h-4 w-4 cursor-pointer opacity-0 absolute peer"
          id="checkbox"
          type="checkbox"
        />
        <div
          class="w-4 h-4 flex flex-shrink-0 justify-center items-center mr-2 peer-checked:bg-primary-clicked peer-checked:border-0 border-neutral-border border cursor-pointer"
        >
          <svg
            fill="none"
            height="8"
            viewBox="0 0 10 8"
            width="10"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M1 4L3.5 6.5L9 1"
              stroke="white"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
          </svg>
        </div>
        <label
          class="select-none text-neutral-text-primary text-13 leading-24 font-medium cursor-pointer"
          for="checkbox"
        >
          Developer
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex items-center"
    >
      <input
        class="h-4 w-4 cursor-pointer opacity-0 absolute peer"
        id="checkbox"
        type="checkbox"
      />
      <div
        class="w-4 h-4 flex flex-shrink-0 justify-center items-center mr-2 peer-checked:bg-primary-clicked peer-checked:border-0 border-neutral-border border cursor-pointer"
      >
        <svg
          fill="none"
          height="8"
          viewBox="0 0 10 8"
          width="10"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1 4L3.5 6.5L9 1"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
          />
        </svg>
      </div>
      <label
        class="select-none text-neutral-text-primary text-13 leading-24 font-medium cursor-pointer"
        for="checkbox"
      >
        Developer
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
